### PR TITLE
Handle case where table-filter is not found

### PR DESCRIPTION
**Pull request by rzane**
When there are no standard personas, the `input[data-behavior=table-filter]` element won't exist (see [here](https://github.com/Betterment/demo_mode/blob/46f82a095c4cd283c93354d7734437d7b4aff3c5/app/views/demo_mode/sessions/new.html.erb#L21)). As a result, the following JS error is thrown:
> TypeError: Cannot set properties of null (setting 'oninput')

This PR does an early return if that input is not found.

In addition to this bugfix, the following maintenance has been performed:
* Bumped the version.
* Switch to Cuprite. It's simpler to configure and would have caught this error with `js_errors: true`.
* Removed Ruby 2.7 from the build matrix.
* Removed Rails 6.0 from the build matrix.
* Added Ruby 3.2 to the build matrix.
* Fixed new Rubocop violations.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -9,24 +9,17 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby: ['2.7', '3.0', '3.1']
+        ruby: ['3.0', '3.1', '3.2']
```
Comment (smudge) [inline .github/workflows/tests.yml:12]: If we're dropping 2.7 from the build, we should update the [minimum required ruby](https://github.com/Betterment/demo_mode/blob/main/demo_mode.gemspec#L19) in the gemspec. (Or, we should keep 2.7 in the build matrix -- point being, if it's supported, it should be tested.)

File: Gemfile [hunk 2]
```diff
@@ -9,9 +9,9 @@ gem 'appraisal', require: false
 gem 'betterlint', require: false
 gem 'capybara', require: false
 gem 'combustion'
+gem 'cuprite'
```
Comment (smudge) [inline Gemfile:12]: 🙌

File: .rubocop.yml [hunk 3]
```diff
@@ -13,3 +13,7 @@ Betterment/ActiveJobPerformable:
 
 Betterment/UnscopedFind:
   Enabled: false
+
+Bundler/OrderedGems:
+  Exclude:
+    - 'gemfiles/*.gemfile'
```
Comment (smudge) [inline .rubocop.yml:19]: 👍

**Review by smudge — DISMISSED** [review]
domain LGTM

**Review by smudge — CHANGES_REQUESTED** [review]
platform TAFN -- just the note about the min supported Ruby in the gemspec -- we should fully decide whether we're dropping 2.7 support.

File: lib/demo_mode/version.rb [hunk 4]
```diff
@@ -1,3 +1,3 @@
 module DemoMode
-  VERSION = '1.1.1'.freeze
+  VERSION = '1.1.2'.freeze
```
[the remark this answers is not in the document]
Comment (rzane) [reply lib/demo_mode/version.rb:2]: Can we merge this and I'll put up another PR to change the gemspec's required_ruby_version and bump the gem's minor version?

[the remark this answers is not in the document]
Comment (rzane) [reply lib/demo_mode/version.rb:2]: On second thought, I'll just bump the minor now.

**Comment by smudge** [comment]
Looks like Rubocop is still running with TargetRubyVersion 2.7 and complaining.

**Review by smudge — APPROVED** [review]
platform LGTM! Thanks for this!

**Review by smudge — APPROVED** [review]
domain LGTM as well!

**Pull request merged**